### PR TITLE
[occm] reuse deleteListeners in EnsureLoadBalancerDeleted

### PR DESCRIPTION
**Pull request by sbueringer**
All the currently maintained binaries are:
* openstack-cloud-controller-manager (occm)

**What this PR does / why we need it**:
There is currently a difference between `deleteListeners` in `ensureOctaviaLoadBalancer` and the implementation in `ensureLoadBalancerDeleted`. The implementation in `ensureLoadBalancerDeleted` also deletes pool members. As this is not required, because a pool deletion also deletes all members (and it does it considerable faster), I would suggest reusing the `deleteListeners` method. This will speed up loadbalancer deletion in case Neutron LBaaS v2 is used.

**Release note**:

```release-note
NONE
```

<sub>Stefan Büringer <stefan.bueringer@daimler.com>, Daimler TSS GmbH, [legal info/Impressum](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md)</sub>

### CODE REVIEW

File: pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go [hunk 1]
```diff
@@ -2756,52 +2754,9 @@ func (lbaas *LbaasV2) ensureLoadBalancerDeleted(ctx context.Context, clusterName
...
-			err := listeners.Delete(lbaas.lb, listener.ID).ExtractErr()
-			if err != nil && !cpoerrors.IsNotFound(err) {
-				return mc.ObserveRequest(err)
-			}
-			mc.ObserveRequest(nil)
-			provisioningStatus, err := waitLoadbalancerActiveProvisioningStatus(lbaas.lb, loadbalancer.ID)
-			if err != nil {
-				return fmt.Errorf("timeout when waiting for loadbalancer to be ACTIVE after deleting listener, current provisioning status %s", provisioningStatus)
-			}
+		if err := lbaas.deleteListeners(loadbalancer.ID, listenerList); err != nil {
```
Comment (lingxiankong) [inline pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go:2758]: Have you tried deleting listener directly without deleting healthmonitor or pool?

Comment (sbueringer) [reply pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go:2758]: I've tried the following:
```
$ neutron lbaas-loadbalancer-delete kubeapi
listener 165725b2-dad7-4a3b-8c86-a153bfc2bde1 is using this loadbalancer

$ neutron lbaas-listener-delete 165725b2-dad7-4a3b-8c86-a153bfc2bde1
Deleted listener(s): 165725b2-dad7-4a3b-8c86-a153bfc2bde1

$ neutron lbaas-loadbalancer-delete kubeapi
pool a164402b-616f-47bd-a81a-a29732644642 is using this loadbalancer

$ neutron lbaas-pool-delete a164402b-616f-47bd-a81a-a29732644642
healthmonitor 00420307-e586-4662-b8a4-4b3ffe3637f6 is using this pool
```

So looks to me like you can delete the listener, but this won't delete the pools. And when you then try to delete the pools you have to delete the healthmonitors first

**Comment by sbueringer** [comment]
@lingxiankong Can you maybe point me to the right place in the test logs to find out why the octavia tests fail? I found:
```
2020-09-30 10:18:06.270570 | ubuntu-bionic | ++ /home/zuul/src/k8s.io/kubernetes/cluster/kubectl.sh get pv
2020-09-30 10:18:06.270639 | ubuntu-bionic | /bin/bash: line 8: /home/zuul/src/k8s.io/kubernetes/cluster/kubectl.sh: No such file or directory
```

File: pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go (hunk 1 above)
Comment (lingxiankong) [reply pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go:2758]: Thanks

**Comment by lingxiankong** [comment]
/override openlab/cloud-provider-openstack-acceptance-test-csi-manila

**Comment by ramineni** [comment]
Thanks.
/lgtm
/approve

**Pull request merged**